msys2 package with pcre2-8 no more working :(

I'm currently using github actions with msys2 (project pvsneslib on github)
last week , everything was working fine with my windows part of my github action, which is:
```
        uses: msys2/setup-msys2@v2
        with:
          update: true
          msystem: UCRT64
          install: >-
            mingw-w64-ucrt-x86_64-toolchain
            mingw-w64-ucrt-x86_64-cmake
            mingw-w64-ucrt-x86_64-pcre2
            mingw-w64-ucrt-x86_64-doxygen
            base-devel
            git
            zip
```
This week, when I launch the action, during compilation of the lib, I have
```
gcc -Wall -O2 -pedantic -D__BUILD_DATE="\"20240302\"" -D__BUILD_VERSION="\"1.0.0\"" -DPCRE2_STATIC  build/helpers.o  build/main.o  build/optimizer.o -lpthread -lpthread -static -lpcre2-posix -lpcre2-8 -liconv -o 816-opt.exe
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x18e): undefined reference to `__imp_pcre2_match_data_free_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x19d): undefined reference to `__imp_pcre2_code_free_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x26e): undefined reference to `__imp_pcre2_compile_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x292): undefined reference to `__imp_pcre2_pattern_info_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x2a9): undefined reference to `__imp_pcre2_match_data_create_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x355): undefined reference to `__imp_pcre2_code_free_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x42d): undefined reference to `__imp_pcre2_match_8'
D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../lib\libpcre2-posix.a(libpcre2_posix_la-pcre2posix.o):(.text+0x4e5): undefined reference to `__imp_pcre2_get_ovector_pointer_8'
collect2.exe: error: ld returned 1 exit status
```
As you can see, pcre2-8 is loaded but the linker seems to have problem :(
Any clew about where to search for a solution?
As I said, it worked perfectly last week.

--- a/mingw-w64-pcre2/PKGBUILD
+++ b/mingw-w64-pcre2/PKGBUILD
@@ -4,7 +4,7 @@ _realname=pcre2
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=10.43
-pkgrel=1
+pkgrel=2
 pkgdesc="A library that implements Perl 5-style regular expressions (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -19,33 +19,59 @@ depends=("${MINGW_PACKAGE_PREFIX}-bzip2"
 source=(https://github.com/PhilipHazel/pcre2/releases/download/${_realname}-${pkgver}/${_realname}-${pkgver}.tar.bz2{,.sig})
 sha256sums=('e2a53984ff0b07dfdb5ae4486bbb9b21cca8e7df2434096cc9bf1b728c350bcb'
             'SKIP')
-validpgpkeys=('45F68D54BBE23FB3039B46E59766E084FB0F43D8')
+validpgpkeys=('45F68D54BBE23FB3039B46E59766E084FB0F43D8') # Philip Hazel <Philip.Hazel@gmail.com>
 
 build() {
-  mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
+  local -a _common_config=(
+    --enable-pcre2-8
+    --enable-pcre2-16
+    --enable-pcre2-32
+    --enable-pcre2grep-libz
+    --enable-pcre2grep-libbz2
+    --enable-pcre2test-libedit
+  )
+
+  mkdir -p "${srcdir}/build-${MSYSTEM}-static" && cd "${srcdir}/build-${MSYSTEM}-static"
+
+  CPPFLAGS+=" -DPCRE2_STATIC" \
+  "${srcdir}"/${_realname}-${pkgver}/configure \
+    --prefix=${MINGW_PREFIX} \
+    --build=${MINGW_CHOST} \
+    --host=${MINGW_CHOST} \
+    --target=${MINGW_CHOST} \
+    --enable-static \
+    --disable-shared \
+    "${_common_config[@]}"
+
+  make
+
+  mkdir -p "${srcdir}/build-${MSYSTEM}-shared" && cd "${srcdir}/build-${MSYSTEM}-shared"
 
   "${srcdir}"/${_realname}-${pkgver}/configure \
     --prefix=${MINGW_PREFIX} \
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
     --target=${MINGW_CHOST} \
-    --enable-pcre2-8 \
-    --enable-pcre2-16 \
-    --enable-pcre2-32 \
-    --enable-pcre2grep-libz \
-    --enable-pcre2grep-libbz2 \
-    --enable-pcre2test-libedit
+    --enable-shared \
+    --disable-static \
+    "${_common_config[@]}"
 
   make
 }
 
 check() {
-  cd "${srcdir}/build-${MSYSTEM}"
+  cd "${srcdir}/build-${MSYSTEM}-static"
+  make check || true
+
+  cd "${srcdir}/build-${MSYSTEM}-shared"
   make check || true
 }
 
 package() {
-  cd "${srcdir}/build-${MSYSTEM}"
+  cd "${srcdir}/build-${MSYSTEM}-static"
+  make DESTDIR="${pkgdir}" install
+
+  cd "${srcdir}/build-${MSYSTEM}-shared"
   make DESTDIR="${pkgdir}" install
 
   # License files

